Lib/test/test_funcattrs.py

```diff
@@ -482,24 +482,41 @@ def test_builtin__qualname__(self):
     @support.cpython_only
     def test_builtin__self__(self):
         # See https://github.com/python/cpython/issues/58211.
-        import time
         import builtins
+        import time
 
         # builtin function:
-        self.assertEqual(len.__self__, builtins)
-        self.assertEqual(time.sleep.__self__, time)
+        self.assertIs(len.__self__, builtins)
+        self.assertIs(time.sleep.__self__, time)
 
         # builtin classmethod:
-        self.assertEqual(dict.fromkeys.__self__, dict)
-        self.assertEqual(float.__getformat__.__self__, float)
+        self.assertIs(dict.fromkeys.__self__, dict)
+        self.assertIs(float.__getformat__.__self__, float)
 
         # builtin staticmethod:
-        self.assertEqual(str.maketrans.__self__, None)
-        self.assertEqual(bytes.maketrans.__self__, None)
+        self.assertIsNone(str.maketrans.__self__)
+        self.assertIsNone(bytes.maketrans.__self__)
 
         # builtin bound instance method:
-        self.assertEqual([1, 2, 3].append.__self__, [1, 2, 3])
-        self.assertEqual({'foo': 'bar'}.pop.__self__, {'foo': 'bar'})
+        l = [1, 2, 3]
+        self.assertIs(l.append.__self__, l)
+
+        d = {'foo': 'bar'}
+        self.assertEqual(d.pop.__self__, d)
+
+        # builtin constants bound method:
+        self.assertFalse(False.__repr__.__self__)
+        self.assertIsNone(None.__repr__.__self__)
+        self.assertTrue(True.__repr__.__self__)
+
+        ni = NotImplemented
+        self.assertIs(ni.__repr__.__self__, ni)
+
+        e = Ellipsis
+        self.assertIs(e.__repr__.__self__, e)
+
+        d = __debug__
+        self.assertIs(d.__repr__.__self__, d)
```

Comment:
`None` is the only interesting case here, because `__self__` is `None` only for static methods and for bound methds of `None`.
```suggestion
        self.assertIsNone(None.__repr__.__self__)
```
All other tests are redundant.